go/types/path.go

```diff
@@ -421,14 +429,30 @@ Switch:
 type TypeAnnotation struct {
 }
 
-func (ann TypeAnnotation) Resolve(v Value) Value {
+func (ann TypeAnnotation) Resolve(v Value, vr ValueReader) Value {
 	return TypeOf(v)
 }
 
 func (ann TypeAnnotation) String() string {
 	return "@type"
 }
 
+// TargetAnnotation is a PathPart annotation to resolve to the targetValue of the Ref it is resolved on.
+type TargetAnnotation struct {
+}
+
+func (ann TargetAnnotation) Resolve(v Value, vr ValueReader) Value {
+	if r, ok := v.(Ref); ok {
+		return r.TargetValue(vr)
```

Comment:
It would be helpful to throw a useful error if `vr` is nil because it seems like an easy gotcha.